src/VisualStudio/Core/Def/NavigateTo/RoslynSearchResultViewFactory.cs

```diff
@@ -64,9 +64,9 @@ public Task<IReadOnlyList<SearchResultPreviewPanelBase>> GetPreviewPanelsAsync(S
                 return null;
 
             Uri? absoluteUri;
-            if (document.IsSourceGeneratedDocument)
+            if (document.SourceGeneratedDocumentIdentity is not null)
             {
-                absoluteUri = ProtocolConversions.CreateUriFromSourceGeneratedFilePath(filePath);
+                absoluteUri = SourceGeneratedDocumentUri.Create(document.SourceGeneratedDocumentIdentity.Value);
             }
             else
             {
```

Comment:
And to be clear if we just pass a URI directly then we can delete this else block too. This was added as a part of #72442 which was a fine tactical fix but shouldn't be needed going forward, I'd imagine.

Reply:
Going to leave this as-is - the navigable document is created at the features layer which doesn't include any of the URI stuff we have at the protocol layer.  So I believe the else is still needed here since we won't be using the GetUri extension on document.

I don't see a problem with moving some of the URI stuff down to features, but I think that is best left to a separate PR (a number of helpers would need to move along with it)